tests/test_testserver.py

```diff
@@ -0,0 +1,128 @@
+import threading
+import socket
+import time
+
+import pytest
+import requests
+from testserver.server import Server
+
+class TestTestServer:
+    def test_basic(self):
+        question = b"sucess?"
+        answer = b"yeah, success"
+        def handler(sock):
+            text = sock.recv(1000)
+            assert text == question 
+            sock.send(answer)
+        
+        with Server(handler) as (host, port):
+            sock = socket.socket()
+            sock.connect((host, port))
+            sock.send(question)
```

Comment:
So here's a useful thing to know: `sock.send()` may not send all the data you pass to it. It has a return value that is how many bytes of the data to sent actually get sent.

You may find it helpful to replace this `send` with `sendall`, and probably all instances of it in this pull request. =)